pkg/compactor/compactor.go

```diff
@@ -311,9 +312,15 @@ func newMultitenantCompactor(
 			Name: "cortex_compactor_runs_completed_total",
 			Help: "Total number of compaction runs successfully completed.",
 		}),
-		compactionRunsFailed: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
-			Name: "cortex_compactor_runs_failed_total",
-			Help: "Total number of compaction runs failed.",
+		compactionRunsErred: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
```

Comment:
We don't need two metrics with const labels. Can be just `NewCounterVec()` with the dynamic `reason` label.

Reply:
i took up that way initially, but had to copy `WithLabelValues("error")` and `WithLabelValues("shutdown")` a few times
* once to `Inc()` it
* once when creating the metrics so we initialize them to 0 
* once in tests

so i decided to just have it in a single place

Reply:
Ok, sounds good.